### PR TITLE
Improve systemd packaging and docs

### PR DESCRIPTION
**Pull request by daemon1024**
- To avoid the confusion of package names, referencing the official guide for BCC
- Add BCC dependency in deb and rpm packages
- Generate archive for manual installation
- Updated docs for manual installation
- Added makefile target for local test releases

### CODE REVIEW

**Comment by nam-jaehyun** [comment]
Can we add bcc as a required package (a part of kubearmor deb or rpm)?
It would be better to automatically install bcc during the installation... I think

**Comment by daemon1024** [comment]
So `dpkg -i` doesn't auto install dependency.
image

We can use `apt install ./...`
image

I will update the docs accordingly

A concern is that it doesn't detect about the BCC installation from source, and will install bpfcc regardless is that fine?
Also for dpkg installation on distributions other than debian/ubuntu the package name is different and will conflict.

**Comment by nyrahul** [comment]
> A concern is that it doesn't detect about the BCC installation from source, and will install bpfcc regardless is that fine? Also for dpkg installation on distributions other than debian/ubuntu the package name is different and will conflict.

I think this is still better. May be we can let user forcefully install in case BCC is manually installed and keep a sentence about it in the documentation?

**Comment by daemon1024** [comment]
We now have a tar archive file as well for manual installation
Inspired from https://github.com/containerd/containerd/blob/main/docs/cri/installation.md#step-2-install-containerd
You can find the files here https://github.com/daemon1024/KubeArmor/releases/tag/v0.3.0

~~I will update the relevant docs when I get some time.~~
~~Let me know when do we plan for the next release will try to wrap it up before that!~~

**Comment by daemon1024** [comment]
Can test out the release at https://github.com/daemon1024/KubeArmor/releases/tag/v0.3.1

**Review by nyrahul — DISMISSED** [review]

File: getting-started/kubearmor_vm.md [hunk 1]
```diff
@@ -4,9 +4,24 @@ This recipe explains how to use KubeArmor directly on VM/Bare-Metal host and was
...
 
-1. Install pre-requisites `sudo apt install bpfcc-tools linux-headers-$(uname -r)`
-2. Download the [latest release of KubeArmor](https://github.com/kubearmor/KubeArmor/releases)
-2. Install KubeArmor `sudo dpkg -i kubearmor_VER_linux-amd64.deb` ... where VER is the kubearmor release version
+1. Download the [latest release of KubeArmor](https://github.com/kubearmor/KubeArmor/releases)
+2. Install KubeArmor `sudo apt install ./kubearmor_${VER}_linux-amd64.deb` ... where VER is the kubearmor release version. This will automatically install the required dependencies.
+
+<details><summary>For distributions other than Ubuntu/Debian</summary>
+<p>
+1. Refer [Installing BCC](https://github.com/iovisor/bcc/blob/master/INSTALL.md#installing-bcc) to install pre-requisites.
```
Comment (nam-jaehyun) [inline getting-started/kubearmor_vm.md:13]: New lines didn't work.

File: KubeArmor/Makefile [hunk 2]
```diff
@@ -89,32 +89,21 @@ ifeq (, $(shell which gosec))
...
-	cd $(CURDIR); VERSION=$(shell git describe --tags --always --dirty) nfpm pkg --packager rpm
-
-.PHONY: systemd
-systemd: systemd-deb systemd-rpm
+	cd $(CURDIR); curl -sfL https://raw.githubusercontent.com/kubearmor/kubearmor-client/main/install.sh | sh -s -- -b .
+	cd $(CURDIR); VERSION=$(shell git describe --tags --always --dirty) goreleaser release --rm-dist --skip-publish --skip-sign --skip-validate
 
 .PHONY: clean
 clean:
 	cd $(CURDIR); sudo rm -f kubearmor /tmp/kubearmor.log
```
Comment (nam-jaehyun) [inline KubeArmor/Makefile]: can you have a single line for kuarmor, /tmp/kubearmor.log and dist?

**Review by nam-jaehyun — COMMENTED** [review]
request minor updates

**Pull request merged**